src/ray/core_worker/context.cc

```diff
@@ -49,7 +52,10 @@ struct WorkerThreadContext {
     // thread), so there's no risk of conflicting put object IDs, either.
     // See https://github.com/ray-project/ray/issues/10324 for further details.
     auto num_returns = current_task_ != nullptr ? current_task_->NumReturns() : 0;
-    return num_returns + ++put_counter_;
+
+    // We reserve max_num_generator_returns_ number of indexes for the generator
+    // return so that all generator return can have consistent ids given an index.
+    return num_returns + max_num_generator_returns_ + ++put_counter_;
```

Comment:
Q: should we hard fail or raise an exception if generator return exceeds this number?

Reply:
I feel we can check failure for now.